CodePush.js

```diff
@@ -354,7 +354,11 @@ async function syncInternal(options = {}, syncStatusChangeCallback, downloadProg
         const dialogButtons = [{
           text: null,
           onPress: async () => {
-            resolve(await doDownloadAndInstall());
```

Comment:
Would it be simpler to just do `doDownloadAndInstall().then(resolve, reject)`? The `Alert` module simply calls the `onPress` function, without regard to the fact that it could return a `Promise`, so it feels a little odd to make the arrow function `async` simply to be able to `await` the download call and propagate a rejection using `try/catch`.